packages/rum-core/src/domain/assembly.ts

```diff
@@ -189,6 +190,14 @@ export function startRumAssembly(
             session.sessionReplay === SessionReplayState.SAMPLED
         }
 
+        if (
+          // TODO: remove ff and should always add anonymous user id
+          isExperimentalFeatureEnabled(ExperimentalFeature.ANONYMOUS_USER_TRACKING) &&
+          !commonContext.user.anonymous_id
+        ) {
+          const { type } = configuration.sessionStoreStrategyType ?? { type: 'LocalStorage' }
+          commonContext.user.anonymous_id = retrieveAnonymousId(type)
```

Comment:
**💬 suggestion:** ‏a bit related to my previous comment, but we should probably be able to do something like:
```suggestion
          commonContext.user.anonymous_id = session.getAnonymousUserId()
```